source/guide/3.2/plugin-aws.md

```diff
@@ -0,0 +1,219 @@
+    * validation for the private key path supplied not to exist if it's a new keypair resource.
+    * validation for the private key path supplied to exist and have the correct permissions and/or owner if it's an existing keypair resource.
+
+**Attributes:**
+
+See the [common Runtime Properties section](#runtime-properties).
+
+
+## cloudify.aws.nodes.SecurityGroup
+
+**Derived From:** [cloudify.nodes.SecurityGroup](reference-types.html)
+
+**Properties:**
+
+  * `description` a description of the security group.
+  * `rules` key-value security_group_rule configuration as described in [http://boto.readthedocs.org/en/latest/ref/ec2.html#boto.ec2.securitygroup.SecurityGroup.authorize](http://boto.readthedocs.org/en/latest/ref/ec2.html#boto.ec2.securitygroup.SecurityGroup.authorize). Defaults to `[]`.
+      * `ip_protocol`
+      * `from_port`
+      * `to_port`
+      * `cidr_ip`
+  * `use_external_resource` a boolean for setting whether to create the resource or use an existing one. See the [using existing resources section](#using-existing-resources). Defaults to `false`.
+  * `resource_id` name to give to the new resource or the name or ID of an existing resource when the `use_external_resource` property is set to `true` (see the [using existing resources section](#using-existing-resources)). Defaults to `''` (empty string).
+
+**Mapped Operations:**
+
+  * `cloudify.interfaces.lifecycle.create` creates the security group, along with its defined rules.
+  * `cloudify.interfaces.lifecycle.delete` deletes the security group.
+  * `cloudify.interfaces.validation.creation` see [common validations section](#Validations). Additionally, the *CIDR* of rules which specify one is verified to be of the correct format.
+
+**Attributes:**
+
+See the [common Runtime Properties section](#runtime-properties).
+
+
+## cloudify.aws.nodes.ElasticIP
+
+**Derived From:** [cloudify.nodes.Root](reference-types.html)
+
+**Properties:**
+
+  * `use_external_resource` a boolean for setting whether to create the resource or use an existing one. See the [using existing resources section](#using-existing-resources). Defaults to `false`.
+  * `resource_id` the IP or ID of an existing floating IP when the `use_external_resource` property is set to `true` (see the [using existing resources section](#using-existing-resources)). Defaults to `''` (empty string).
+
+**Mapped Operations:**
+
+  * `cloudify.interfaces.lifecycle.create` creates the elastic IP
+  * `cloudify.interfaces.lifecycle.delete` deletes the elastic IP
+  * `cloudify.interfaces.validation.creation` see [common validations section](#Validations).
+
+**Attributes:**
+
+See the [common Runtime Properties section](#runtime-properties).
+
+Note that the actual IP is available via the `elastic_ip` runtime-property.
+
+
+
+# Relationships
+
+In general, relationships represent node dependenies within Cloudify.
```

Comment:
Please reference the terminology page when you want to explain a specific term (e.g. in this case, relationships).

Thanks!